kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ChecksumLogReplayMetricsTestBase.scala

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults
+import java.io.File
+import java.nio.file.Files
+
+import io.delta.kernel.Table
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.util.FileNames
+
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
+
+/**
+ * Base trait for testing log replay optimizations when reading tables with checksum files.
+ * This trait contains common test setup and test cases but allows specific metadata types
+ * to customize how they load and verify data.
+ *
+ * Test subclasses implement specialized behavior for loading different items:
+ * - PandMCheckSumLogReplayMetricsSuite - tests Protocol and Metadata loading
+ * - DomainMetadataCheckSumReplayMetricsSuite - tests Domain Metadata loading
+ */
+trait ChecksumLogReplayMetricsTestBase extends LogReplayBaseSuite {
+
+  /////////////////////////
+  // Test Helper Methods //
+  /////////////////////////
+
+  /**
+   * Creates a test table with checksum files.
+   * Produces a table with versions 0 to 11 including .json files, .crc files,
+   * and a checkpoint at version 10.
+   *
+   * @param path Path where the table should be created
+   */
+  def buildTableWithCrc(path: String): Unit = {
+    withSQLConf(DeltaSQLConf.DELTA_WRITE_CHECKSUM_ENABLED.key -> "true") {
+      spark.sql(
+        s"CREATE TABLE delta.`$path` USING DELTA AS " +
+          s"SELECT 0L as id")
+      for (_ <- 0 to 10) { appendCommit(path) }
```

Comment:
just add a require saying there is a checkpoint. In future if we update the default checkpoint interval the test throws error,